### PR TITLE
Add Diurnal Temp Variation Endpoint for Arctic EDS

### PR DESCRIPTION
**Pull request by charparr**
This PR adds an endpoint for mean annual diurnal temperature variation (°C).  You can think of this as the magnitude of the day-night temperature swing you'd experience on average for a year, for a given location and time. The endpoints in this PR are min-mean-max summarized (except for /all/) but diurnal_temp_variation is a top-level route. To test this PR, try a few different queries after setting your Rasdaman backend to Apollo:

http://localhost:5000/diurnal_temp_variation/projected/point/62.0628/-150

http://localhost:5000/diurnal_temp_variation/historical/point/68/-149

http://localhost:5000/diurnal_temp_variation/hp/point/62/-166.1627

http://localhost:5000/diurnal_temp_variation/all/point/69/-151

http://localhost:5000/diurnal_temp_variation/projected/point/65/-145/2070/2099

http://localhost:5000/eds/diurnal_temp_variation/point/65/-147

Also try a CSV query and verify that the CSV makes sense.

This endpoint is very similar to `wet_days_per_year`.

### CODE REVIEW

File: routes/diurnal_temp_variation.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+            composed as "y1:y2" for bbox query, where y1 and y2 are
+            lower and upper bounds of bbox
+        cov_id (str): Rasdaman coverage ID
+        models (str): Comma-separated numbers of requested models
+        years (str): Colon-separated full date-time i.e.
+            "\"2006-01-01T00:00:00.000Z\":\"2100-01-01T00:00:00.000Z\""
+        summary_stat(int): Integer between 0-2 where:
+            - 0 = mean summary
+            - 1 = maximum summary
+            - 2 = minimum summary
```
Comment (cstephen) [inline routes/diurnal_temp_variation.py:53]: These comments don't seem to agree with the code below.

File: routes/diurnal_temp_variation.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+        min_year = years_lu["historical"]["min"]
+        max_year = years_lu["historical"]["max"]
+        timestring = (
+            f'"{min_year}-01-01T00:00:00.000Z":"{max_year}-01-01T00:00:00.000Z"'
+        )
+        if start_year is not None:
+            timestring = (
+                f'"{start_year}-01-01T00:00:00.000Z":"{end_year}-01-01T00:00:00.000Z"'
+            )
+        for summary_stat in range(0, 3):
```
Comment (cstephen) [inline routes/diurnal_temp_variation.py:227]: How does this relate to the prior mismatch in the prior documentation?

File: templates/diurnal_temp_variation/point.html [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td><a href="https://catalog.snap.uaf.edu/geonetwork/srv/eng/catalog.search#/metadata/7825535c-edff-4a82-89f3-9183e6cb2b42">Historical downscaled ERA-Interim data (20 km)</a></td>
+    <td><a href="https://catalog.snap.uaf.edu/geonetwork/srv/eng/catalog.search#/metadata/7825535c-edff-4a82-89f3-9183e6cb2b42"> Projected Dynamically Downscaled Climate Data for the State of Alaska and surrounding regions at 20km spatial resolution and hourly temporal resolution</a></td>
+    <td>RCP 8.5</td>
+    <td>GFDL-CM3, NCAR-CCSM4</td>
+    <td>1980-2009</td>
+    <td>2006-2100</td>
```
Comment (cstephen) [inline templates/diurnal_temp_variation/point.html:150]: Use `&ndash;` here instead of hyphen.

File: templates/diurnal_temp_variation/point.html [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    "madtrmean": &lt;the mean of the mean annual diurnal temperature variation (°C) across all models and years used for the summary era time period>,
+    "madtrmin": &lt;the min of the mean annual diurnal temperature variation (°C) across all models and years used for the summary era time period>
+  },
+  ...
+}
+</pre>
+
+<h4>Comprehensive queries</h4>
+<p>
+  The following query will fetch a complete mean annual diurnal temperature variation (°C) set of data composed of the ERA-Interim historical baseline (1980-2009) and the entire projection range (2006-2100) for both available models, GFDL-CM3 and NCAR-CCSM4 (both RCP 8.5).
```
Comment (cstephen) [inline templates/diurnal_temp_variation/point.html:76]: Use `&ndash;` and `&deg;` here.

**Review by cstephen — CHANGES_REQUESTED** [review]
Things look great overall! I've added a few code comments about minor typography and code comment issues.

**Comment by charparr** [comment]
This PR has become super stale. We're going to consider re-baking extreme diurnal temperature swings into our climate indicator development that will parallel our CMIP6 work. They might be relevant for materials science and engineering (thermal expansion / contraction). However, no one has asked for this specific indicator, the development here was initiated by the presence of this dataset in the original Environmental Atlas.

**Pull request closed without merging**